code/modules/mob/living/simple_animal/hostile/ordeal/gold.dm

```diff
@@ -0,0 +1,801 @@
+					continue
+				area_of_effect |= T
+	if (!LAZYLEN(area_of_effect))
+		return
+	can_act = FALSE
+	dir = dir_to_target
+	playsound(get_turf(src), 'sound/weapons/fixer/generic/sheath2.ogg', 75, 0, 5)
+	for(var/turf/T in area_of_effect)
+		new /obj/effect/temp_visual/cult/sparks(T)
+	SLEEP_CHECK_DEATH(0.8 SECONDS)
+	var/slash_damage = 35
+	playsound(get_turf(src), 'sound/weapons/fixer/generic/blade3.ogg', 100, 0, 5)
+	for(var/turf/T in area_of_effect)
+		new /obj/effect/temp_visual/smash_effect(T)
+		for(var/mob/living/L in T)
+			if(faction_check_mob(L))
+				continue
+			if (L == src)
+				continue
+			L.apply_damage(slash_damage, PALE_DAMAGE, null, L.run_armor_check(null, PALE_DAMAGE), spread_damage = TRUE)
+	SLEEP_CHECK_DEATH(0.5 SECONDS)
+	can_act = TRUE
+
+/mob/living/simple_animal/hostile/ordeal/white_lake_corrosion/bullet_act(obj/projectile/P)
+	visible_message("<span class='userdanger'>[P] is easily deflected by [src]!</span>")
+	P.Destroy()
+	return
+
+/mob/living/simple_animal/hostile/ordeal/white_lake_corrosion/attacked_by(obj/item/I, mob/living/user)
+	var/checkdir = check_target_facings(user, src)
+	if((get_dist(user, src) > 1) || checkdir == FACING_EACHOTHER)
+		visible_message("<span class='userdanger'>[user]'s attack is easily deflected by [src]!</span>")
+		return
+	return ..()
+
+/mob/living/simple_animal/hostile/ordeal/sin_gloom
+	name = "Peccatulum Morositatis"
+	desc = "An insect-like entity with a transparant body."
+	icon = 'ModularTegustation/Teguicons/48x48.dmi'
+	icon_state = "sinflea"
+	icon_living = "sinflea"
+	icon_dead = "flea_dead"
+	faction = list("gold_ordeal")
+	maxHealth = 300
+	health = 300
+	melee_damage_type = WHITE_DAMAGE
+	armortype = WHITE_DAMAGE
+	rapid_melee = 2
+	melee_damage_lower = 14
+	melee_damage_upper = 14
+	attack_verb_continuous = "bashes"
+	attack_verb_simple = "bashes"
+	attack_sound = 'sound/effects/ordeals/gold/flea_attack.ogg'
+	deathsound = 'sound/effects/ordeals/gold/flea_dead.ogg'
+	damage_coeff = list(BRUTE = 1, RED_DAMAGE = 2, WHITE_DAMAGE = 0.8, BLACK_DAMAGE = 0.8, PALE_DAMAGE = 1)
+	butcher_results = list(/obj/item/food/meat/slab/human/mutant/slime = 1)
+	guaranteed_butcher_results = list(/obj/item/food/meat/slab/human/mutant/slime = 1)
+	is_flying_animal = TRUE
+	ranged = TRUE
+	minimum_distance = 2 // Don't move all the way to melee
+	projectiletype = /obj/projectile/beam/water_jet
+	projectilesound = 'sound/effects/ordeals/gold/flea_attack.ogg'
+
+/mob/living/simple_animal/hostile/ordeal/sin_gloom/MeleeAction()
+	if(health <= maxHealth*0.5 && stat != DEAD)
+		walk_to(src, 0)
+		animate(src, transform = matrix()*1.8, time = 15)
+		addtimer(CALLBACK(src, .proc/DeathExplosion), 15)
+	..()
+
+/mob/living/simple_animal/hostile/ordeal/sin_gloom/proc/DeathExplosion()
+	if(QDELETED(src))
+		return
+	visible_message("<span class='danger'>[src] suddenly explodes!</span>")
+	playsound(loc, 'sound/effects/ordeals/gold/tentacle_explode.ogg', 60, TRUE)
+	for(var/mob/living/L in view(2, src))
```

Comment:
```suggestion
	for(var/mob/living/L in viewers(2, src))
```
This one *may* be more efficient as it simply checks mobs instead of all objects in view. Uncertain.